WeatherForecast/WeatherForecast/New Group/ViewControllers/WeatherForecastViewController.swift

```diff
@@ -0,0 +1,128 @@
+import UIKit
+import CoreLocation
+
+final class WeatherForecastViewController: UIViewController {
+    private let weatherForecastView = WeatherForecastView()
+    
+    private let locator = Locator()
+    private var networker: Networker?
+    
+    private var currentWeathermodel: Model.CurrentWeather?
+    private var fiveDaysWeatherModel: Model.FiveDaysWeather?
+    private var coordinate: CLLocationCoordinate2D?
+    private var placemark: CLPlacemark?
+    
+    override func loadView() {
+        view = weatherForecastView
+        
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        weatherForecastView.collectionView.delegate = self
+        weatherForecastView.collectionView.dataSource = self
+        
+        configureWeatherData()
+        configureRefreshControl()
+    }
+    
+    private func configureWeatherData() {
+        let group = DispatchGroup()
+        
+        group.enter()
+        DispatchQueue.global(qos: .userInteractive).async {
+            self.locator.requestData { coordinate, placemark in
```

Comment:
이번에 수정은 하지 않으셔도 되긴하는데요. 이런 형태의 코드는 좀 위험해요. 내부에서 무조건 callback을 전달하는 구조가 아니면 여기서 group은 끝나지 않을수있어요. 다음 스텝에서? 혹은 다음에 코드를 작성할때 위 coordinate를 보장된 후 group의 코드가 진행한느 형태면 좋을거같아요.

Reply:
위치 권한이 없는 경우에는 기본적으로 설정된 특정 지역의 날씨 정보를 띄워지게 했습니다. 예전에 카카오맵 API를 사용했을 때, 기본적으로 띄워지는 곳이 카카오 본사였던 것에서 착안하여, 기본값으로 사용할 수 있는 `coordinate`를 지정했습니다.

다만, 이러한 방향이 UX적으로 올바른지 고민됩니다.

또한, `didFindLocation`이 추가된 이유는 refresh 시 `locationManager(didUpdateLocations:)`이 2번 호출되는 경우가 있어 앱이 crash나는 경우가 있어 불가피하게 flag 변수를 만들었습니다.

참고. [해당 Trouble Shooting에 대한 정리](https://www.notion.so/tastycode/e4852bcf2b0b47b0b28efcc34d5df294?pvs=4#3a3fe167faaa4087ba309d9293dc96a7)

```swift
// LocationManager.swift
  func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
        if didFindLocation {
            didFindLocation = false
            fetchPlacemark(for: CLLocation(latitude: DefaultLocation.latitude, longitude: DefaultLocation.longitude))
            locationCompletion?(.failure(LocationError.didFailFetchLocationError))
        }
    }
```
```suggestion
            self.locator.requestData { coordinate, placemark in
```